contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,672 @@
+        string moniker;
+        string identity;
+        string website;
+        string details;
+    }
+
+    struct Commission {
+        uint256 rate; // the commission rate charged to delegators
+        uint256 maxRate; // maximum commission rate which validator can ever charge
+        uint256 maxChangeRate; // maximum daily increase of the validator commission
+    }
+
+    struct SlashRecord {
+        uint256 slashAmount;
+        uint256 slashHeight;
+        uint256 jailUntil;
+        SlashType slashType;
+    }
+
+    enum SlashType {
+        DoubleSign,
+        DownTime,
+        MaliciousVote
+    }
+
+    enum UpdateDirection {
+        Up,
+        Down
+    }
+
+    /*----------------- events -----------------*/
+    event ValidatorCreated(address indexed validator, address indexed operator, address indexed poolModule, bytes voteAddress);
+    event CommissionRateEdited(address indexed validator, uint256 commissionRate);
+    event OperatorEdited(address indexed validator, address indexed newOperator);
+    event VoteAddressEdited(address indexed validator, bytes newVoteAddress);
+    event DescriptionEdited(address indexed validator);
+    event Delegated(address indexed validator, address indexed delegator, uint256 bnbAmount);
+    event Undelegated(address indexed validator, address indexed delegator, uint256 bnbAmount);
+    event Redelegated(address indexed srcValidator, address indexed dstValidator, address indexed delegator, uint256 bnbAmount);
+    event ValidatorSlashed(address indexed validator, uint256 slashAmount, uint256 slashHeight, uint256 jailUntil, SlashType slashType);
+    event ValidatorJailed(address indexed validator);
+    event ValidatorUnjailed(address indexed validator);
+    event Claimed(address indexed validator, address indexed delegator, uint256 bnbAmount);
+    event SecurityDepositClaimed(address indexed validator, uint256 sharesAmount);
+    event StakingPaused();
+    event StakingResumed();
+    event paramChange(string key, bytes value);
+
+    /*----------------- modifiers -----------------*/
+    modifier onlyInitialized() {
+        require(_initialized != 0, "NOT_INITIALIZED");
+        _;
+    }
+
+    modifier validatorExist(address validator) {
+        require(validators[validator].poolModule != address(0), "VALIDATOR_NOT_EXIST");
+        _;
+    }
+
+    modifier validatorNotJailed(address validator) {
+        require(!validators[validator].jailed, "VALIDATOR_JAILED");
+        _;
+    }
+
+    modifier onlyOperator(address validator) {
+        require(validators[validator].poolModule != address(0), "VALIDATOR_NOT_EXIST");
+        if (msg.sender != validator) {
+            require(msg.sender == validators[validator].operatorAddress, "NOT_OPERATOR");
+        }
+        _;
+    }
+
+    modifier whenNotPaused() {
+        require(!_stakingPaused, "STAKE_STOPPED");
+        _;
+    }
+
+    /*----------------- init -----------------*/
+    function initialize() public {
```

Comment:
`initialize` can be done multiple times now, please avoid this.

Reply:
Done